### PR TITLE
ISPN-16221 Add a Version method that is stable across builds

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-16221

### CODE REVIEW

File: commons/all/src/main/java/org/infinispan/commons/util/Version.java [hunk 1]
```diff
@@ -162,6 +164,10 @@ public static short getVersionShort(String versionString) {
       return encodeVersion(a, b, c);
    }
 
+   public static String getUpstreamVersion() {
```
Comment (karesti) [inline commons/all/src/main/java/org/infinispan/commons/util/Version.java:167]: It's ok to me, but having any reference to redhat in the code is accepted ? @tristantarrant

**Review by karesti — COMMENTED** [review]
this is ok for me, just want to double check concerning the redhat ref here

File: commons/all/src/main/java/org/infinispan/commons/util/Version.java (hunk 1 above)
Comment (tristantarrant) [reply commons/all/src/main/java/org/infinispan/commons/util/Version.java:167]: I'd much rather have a `getUnbrandedVersion()` method. The properties file should contain two properties and it should be up to the productization script to handle the values, not the code.

**Review by tristantarrant — CHANGES_REQUESTED** [review]

**Review by tristantarrant — APPROVED** [review]
LGTM

**Pull request merged**